### PR TITLE
Sort companies by date last updated on home page

### PR DESCRIPTION
**Pull request by weedySeaDragon**
#### PT URL: https://www.pivotaltracker.com/story/show/170472179

The default sort order for Companies on the home page (= `companies#index`) is set to sort by the date a Company **attribute** was **last updated** (most recently updated = at the top).

That part was pretty easy.  

But the next step is to also track changes are made to information associated with a Company.  Ex:
   - A member adds an address to the company.  Users would rightly expect that this would count as a change to information about a Company, so that Company should now be sorted at the top of the list
   - likewise with changes to other info associated with a company (= any `has_many` .. relationships):
      - pictures
      - Dinkurs events
   - and what about these:
      - changes to members associated with a company (new ones added for newly approved applications; members removed because of expired membership terms)?
      - changes to business categories associated with a company (new ones added for newly approved applications; business categories removed because of expired membership terms)

I wrote a scenario for adding an address and expecting the Company to show up first in the list (because the address was added): `company-list-order.feature` "Add address to a company (info associated with a company; not an attribute)"  _This scenario currently fails (as expected since I haven't implemented this yet)._ 

Should we break this into 2 different tasks so we can deploy at least something quickly?
1.  Sort companies based only on the `updated_at` attribute (what this PR does right now)
2. Include the associated objects when determining when a company was last updated, i.e. pictures, events, members, business categories

## Changes proposed in this pull request:
1. added a method to set the `.order` for the list of all companies in the `CompaniesController` `index` action (method).  Right now it is just hard-coded to return the clause to order by `updated_at  DESC 
2.
3. Re-organized some features:  Moved most company-related features into the features/companies dir
   - Added a (real) feature name to the company features I moved

---
## Ready for review:
@thesuss @patmbolger @pjbelo

### CODE REVIEW

**Comment by weedySeaDragon** [comment]
@thesuss  gave the 👍  to merge this PR -- sorting the list by date last updated.
Other work should be other PT tasks + future PRs

File: app/controllers/companies_controller.rb [hunk 1]
```diff
@@ -23,17 +23,23 @@ def index
...
+    # put the companies in the desired order
+    @order_companies_by = company_order
+
     # only select companies that are 'complete'; see the Company.complete scope
 
     @all_companies = @search_params.result(distinct: true)
                          .includes(:business_categories)
                          .includes(addresses: [:region, :kommun])
                          .joins(addresses: [:region, :kommun])
+                         .order(@order_companies_by)
```
Comment (patmbolger) [inline app/controllers/companies_controller.rb]: Were you intending to use the `@order_companies_by` instance var in a view?  If not, it is not needed.

File: features/view_all_companies.feature [hunk 2]
```diff
@@ -162,6 +165,7 @@ Feature: Visitor sees all companies
     Given the date is set to "2017-10-01"
     Given I am Logged out
     And I am on the "landing" page
+    And show me the page
```
Comment (patmbolger) [inline features/view_all_companies.feature]: Is this for debugging?  (probably don't want it in production code).

**Review by patmbolger — COMMENTED** [review]
Please see 2 comments.

File: features/view_all_companies.feature (hunk 2 above)
Comment (weedySeaDragon) [reply features/view_all_companies.feature]: doh!

File: app/controllers/companies_controller.rb (hunk 1 above)
Comment (patmbolger) [reply app/controllers/companies_controller.rb]: @weedySeaDragon Did you see my comment re this unneeded(?) instance var?

Comment (weedySeaDragon) [reply app/controllers/companies_controller.rb]: @patmbolger I did see your comment and took it out in my latest commit.  https://github.com/AgileVentures/shf-project/pull/763/commits/0ff2eb0cee8c216a1e44ad35d86cba66679a6ad0
 Maybe you need to pull that latest commit?

**Review by patmbolger — APPROVED** [review]
LGTM

**Pull request merged**